### PR TITLE
Fixes #591: typos, adding the type attribute to lists, and moving the name attribute for some examples

### LINKED ISSUE #591: Fix documentation formatting
Testing the documentation on the staging environment of docs.microsoft.com, there are a couple things to fix:

- lists elements need a type attribute present, to be rendered properly
- OGD, and Poisson regressor had the name attribute misplaced.

### PR DESCRIPTION
**Pull request by sfilipi**
Testing the documentation in the doc.microsoft.com staging environment, i noticed a few typos, unescaped apostrophes, and missing xml tag attributes. 

Fixes #591: typos, adding the type to lists, and fixing the name attribute in OGD and Poisson

### CODE REVIEW

File: src/Microsoft.ML.StandardLearners/Standard/Online/doc.xml [hunk 1]
```diff
@@ -13,8 +13,8 @@
         and an option to update the weight vector using the average of the vectors seen over time (averaged argument is set to True by default).
       </remarks>
     </member>
-    <example>
-      <example name="OGD">
+    <example name="OGD">
```
Comment (Zruty0) [inline src/Microsoft.ML.StandardLearners/Standard/Online/doc.xml]: >example [](start = 5, length = 7)

is this not an error to have nested examples?

Comment (Zruty0) [reply src/Microsoft.ML.StandardLearners/Standard/Online/doc.xml]: Yea I see it's used everywhere. I wonder why though?

---
In reply to: [205926076](https://github.com/dotnet/machinelearning/pull/592#discussion_r205926076) [](ancestors = 205926076)

Comment (Ivanidzo4ka) [inline src/Microsoft.ML.StandardLearners/Standard/Online/doc.xml]: >example name="OGD" [](start = 5, length = 18)

out of curiosity, is order of having name / not having name matter? In AP example you have <example><example name=""> and here you have <example name=""><example>

File: src/Microsoft.ML.Transforms/doc.xml [hunk 2]
```diff
@@ -7,8 +7,7 @@
         Encodes the categorical variable with hash-based encoding. 
       </summary>
       <remarks>
-        CategoricalHashOneHotVectorizer converts a categorical value into an indicator array by hashing the
-        value and using the hash as an index in the bag.
+        CategoricalHashOneHotVectorizer converts a categorical value into an indicator array by hashing the value and using the hash as an index in the bag.
```
Comment (Zruty0) [inline src/Microsoft.ML.Transforms/doc.xml]: >the value [](start = 104, length = 9)

why removing the newline here, but not, say, in line 33?

File: src/Microsoft.ML.StandardLearners/Standard/Online/doc.xml (hunk 1 above)
Comment (sfilipi) [reply src/Microsoft.ML.StandardLearners/Standard/Online/doc.xml]: it is legal in xml. 
The internal example node is needed by the documentation to display the snippet as an example. 

I needed a named wrapper to get the node, and ...just called it example. 

---
In reply to: [205926101](https://github.com/dotnet/machinelearning/pull/592#discussion_r205926101) [](ancestors = 205926101,205926076)

Comment (sfilipi) [reply src/Microsoft.ML.StandardLearners/Standard/Online/doc.xml]: it depends how you include the node in the code. 

see the long answer to Pete's comment. Should clarify it. 
Note the * notation:

<include file='../Microsoft.ML.StandardLearners/Standard/doc.xml' path='doc/members/member[@name="OLS"]/*' />


---
In reply to: [205926135](https://github.com/dotnet/machinelearning/pull/592#discussion_r205926135) [](ancestors = 205926135)

File: src/Microsoft.ML.Transforms/doc.xml (hunk 2 above)
Comment (sfilipi) [reply src/Microsoft.ML.Transforms/doc.xml]: I am not sure why the text is broken in newlines identically how my text is structured for the page corresponding to the CategoricalHashOneHotVectorizer, and it is flowing differently for the CategoricalOneHotVectorizer, disregarding line indentations in the XML. 

Could be because the text in the other one is inside the <para> tags; and we should always use the para tags to keep the convenience formatting in the xml. 

links if curious:
[https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.ml.transforms.categoricalonehotvectorizer?view=ml-dotnet&branch=smoke-test-preview](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.ml.transforms.categoricalonehotvectorizer?view=ml-dotnet&branch=smoke-test-preview)

[https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.ml.transforms.categoricalhashonehotvectorizer?view=ml-dotnet&branch=smoke-test-preview](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.ml.transforms.categoricalhashonehotvectorizer?view=ml-dotnet&branch=smoke-test-preview)






---
In reply to: [205926140](https://github.com/dotnet/machinelearning/pull/592#discussion_r205926140) [](ancestors = 205926140)

File: src/Microsoft.ML.Transforms/doc.xml [hunk 3]
```diff
@@ -33,16 +32,16 @@
         <para>The CategoricalOneHotVectorizer transform passes through a data set, operating on text columns, to
         build a dictionary of categories.
         For each row, the entire text string appearing in the input column is defined as a category.
-        The output of this transform is an indicator vector.
+        The output of this transform is an indicator vector.</para>
         Each slot in this vector corresponds to a category in the dictionary, so its length is the size of the built dictionary.
```
Comment (sharwell) [inline src/Microsoft.ML.Transforms/doc.xml]: :memo: This line should be in a paragraph element.

File: src/Microsoft.ML.FastTree/doc.xml [hunk 4]
```diff
@@ -157,28 +157,28 @@
       </summary>
       <remarks>
         In machine learning​ it is a pretty common and powerful approach to utilize the already trained model in the process of defining features.
-        <para>One such example would be the use of model's scores as features to downstream models. For example, we might run clustering on the original features, 
+        <para>One such example would be the use of model&apos;s scores as features to downstream models. For example, we might run clustering on the original features,
```
Comment (sharwell) [inline src/Microsoft.ML.FastTree/doc.xml]: ❔ Why is this required? It's awkward to read and seems like it would be easy to regress.

File: src/Microsoft.ML.Data/Transforms/doc.xml [hunk 5]
```diff
@@ -49,6 +49,52 @@
...
+      </remarks>
+      <seealso cref=" Microsoft.ML.Runtime.Data.MetadataUtils.Kinds.HasMissingValues"/>
+      <seealso cref="T:Microsoft.ML.Data.DataKind"/>
+    </member>
+    <example name="NAHandle">
+      <example>
+        <code language="csharp">
+          pipeline.Add(new MissingValueHandler(&quot;FeatureCol&quot;, &quot;CleanFeatureCol&quot;)
+          {
+          ReplaceWith  = NAHandleTransformReplacementKind.Mean
```
Comment (sharwell) [inline src/Microsoft.ML.Data/Transforms/doc.xml]: :bulb: Should likely be indented

File: src/Microsoft.ML.Data/Transforms/doc.xml [hunk 6]
```diff
@@ -49,6 +49,52 @@
...
+          It can be changed to true for known length vectors, but it results in an error if changed to false for variable length vectors.
+        </para>
+      </remarks>
+      <seealso cref=" Microsoft.ML.Runtime.Data.MetadataUtils.Kinds.HasMissingValues"/>
+      <seealso cref="T:Microsoft.ML.Data.DataKind"/>
+    </member>
+    <example name="NAHandle">
+      <example>
+        <code language="csharp">
+          pipeline.Add(new MissingValueHandler(&quot;FeatureCol&quot;, &quot;CleanFeatureCol&quot;)
```
Comment (sharwell) [inline src/Microsoft.ML.Data/Transforms/doc.xml]: ❔ Why is the `&quot;` escape needed? I would only expect this if it appeared as the value of an XML attribute.

File: src/Microsoft.ML.Data/Transforms/doc.xml [hunk 7]
```diff
@@ -49,6 +49,52 @@
...
+          The output column can also optionally include an indicator vector for which slots were missing in the input column.
+          This can be done only when the indicator vector type can be converted to the input column type, i.e. only for numeric columns.
+        </para>
+        <para>
+          When computing the mean/max/min value, there is also an option to compute it over the whole column instead of per slot.
+          This option has a default value of true for variable length vectors, and false for known length vectors.
+          It can be changed to true for known length vectors, but it results in an error if changed to false for variable length vectors.
+        </para>
+      </remarks>
+      <seealso cref=" Microsoft.ML.Runtime.Data.MetadataUtils.Kinds.HasMissingValues"/>
```
Comment (sharwell) [inline src/Microsoft.ML.Data/Transforms/doc.xml]: ❔ Is this missing a `T:` prefix? It's unclear why the others are given in full form but I noticed this one is different.
